README.md

```diff
@@ -1,134 +1,121 @@
-    , optionDecoder : Json.Decode.Decoder a
-    }
 ```
 
-5. Viewing the select. Call `SingleSelect.view`. **_Note:_** The `view` for the multi select expects different arguments than seen here. Please refer to the `SmartSelectMulti` documentation for further information.
+5. View the select. Call `Select.view`. **_Note:_** The `view` functions for each of the modules expect slightly different arguments from one another. Please refer to the relevant module documentation for further information.
 
-`SingleSelect.view` expects the following (in order):
-- a boolean indicating if the select is disabled or not
-- the currently selected entity itself
-- the configured settings
-- the smart select instance
+```elm
+view : Model -> Html Msg
+view model =
+    div []
+        [ ...
+        , div
+            [ style "width" "500px" ]
+            [ Select.view False model.products model.select ]
+        ]
+```
 
-6. Updating the select. Here is where we handle the `Msg` we defined in step 3. As indicated before, the message carries with it a tuple containing the `SingleSelect.Msg` for updating the select component as well as the current selection state. 
+6. Update the select. Here is where we handle the `Msg` we defined in step 2. As indicated before, the message carries with it a `Select.Msg` for updating the select component. 
 
 ```elm
 update : Msg -> Model -> ( Model, Cmd Msg )
 update msg model =
     case msg of
         ...
 
-        HandleSelectUpdate ( sMsg, selectedProduct ) ->
+        HandleSelectUpdate sMsg ->
             let
                 ( updatedSelect, selectCmd ) =
-                    SmartSelect.update sMsg (singleSelectSettings model.selectableProducts) selectedProduct model.singleSelect
+                    SmartSelect.update sMsg model.singleSelect
             in
-            ( { model | singleSelect = updatedSelect, selectedProduct = selectedProduct }, selectCmd )
+            ( { model | singleSelect = updatedSelect }, selectCmd )
 ```
 
-There are two things happening here:
-- We always update the selection state in the model based on the selection sent through the tuple. This does not necessarily mean a new selection has occurred.
-- We update the smart select instance with the select `msg` received in the tuple. The `update` function takes the following arguments (in order):
-    - the select msg
-    - the select settings
-    - the *new* selection. **_This is important_**. Always call the update with the selection state received from the tuple.
-    - the select instance to update
-
 `SingleSelect.upate` returns an updated smart select instance and a cmd.
 
-7. The select module uses a subscription to determine when to close (outside of a selection). Wire the picker subscription like below.
+You might be wondering were the selected state is. The smart select stores the select state and exposes a `selected` method to retrieve it. Call this method when you need the selected entity/entities.
```

Comment:
I am still not sure I like this. It sounds a bit unidiomatic. It means whenever you want to do something like send data to the server you need to remember to ask every SmartSelect for its selection.

Reply:
I feel like the benefits outweigh the cons here:

- one less field in model (vs select and selected)
- one less update case to handle (vs internal update msg and selected msg). In the case of the work we are doing, this would save us quite a bit of additional update logic due to the number of selects on the model.

Seeing as the user will need to encode their request data anyway there isn't a huge difference between

```elm
{ ...
, selected : model.selected
...
}
```

and 

```elm
{ ...
, selected : SmartSelect.selected model.selected
...
}
```

And if they need to have actions triggered on select, all they have to do is compare the selection state prior to and after the update.

Reply:
You are right, the decoder can also just call the selected function on the SmartSelect.

Reply:
Yep, those are super paraphrased encoders in the code blocks there, but you got the idea anyway. 😜